packages/calendar-utils/src/calendar-body/calendar-body.styles.js

```diff
@@ -26,20 +26,58 @@ const getClearSectionStyles = (props) => {
     }
   `;
 };
+const getIconBorderColor = (vars, props, state) => {
+  if (props.isDisabled) {
+    return vars.borderColorForInputWhenDisabled;
+  }
+  if (props.hasError) {
+    return vars.borderColorForInputWhenError;
+  }
+  if (props.hasWarning) {
+    return vars.borderColorForInputWhenWarning;
+  }
+  if (props.isReadOnly) {
+    return vars.borderColorForInputWhenReadonly;
+  }
+  if (props.isOpen || state.isFocused) {
+    return vars.borderColorForInputWhenFocused;
+  }
+  return vars.borderColorForInput;
+};
+const getIconFontColor = (vars, props) => {
+  if (props.isDisabled) {
+    return vars.fontColorForInputWhenDisabled;
+  }
+  if (props.hasError) {
+    return vars.fontColorForInputWhenError;
+  }
+  if (props.hasWarning) {
+    return vars.fontColorForInputWhenWarning;
+  }
+  if (props.isReadOnly) {
+    return vars.fontColorForInputWhenReadonly;
+  }
+  return 'initial';
+};
 const getCalendarIconContainerStyles = (props, state) => {
   const vars = {
     ...customProperties,
     ...props.theme,
   };
 
-  const baseIconStyles = css`
+  return css`
     align-items: center;
     box-sizing: border-box;
-    background: none;
+    background-color: ${props.isDisabled
```

Comment:
This should be removed (`none` or `transparent`), because the background is given by the parent container. Otherwise, the background "paints" on top of the box-shadow focus indicator.